persistence-api/src/main/java/io/stargate/db/PersistenceActivator.java

```diff
@@ -0,0 +1,32 @@
+package io.stargate.db;
+
+import io.stargate.core.activator.BaseActivator;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public abstract class PersistenceActivator extends BaseActivator {
+
+  private String BASEDIR = System.getProperty("stargate.basedir");
+
+  public PersistenceActivator(String activatorName) {
+    super(activatorName);
+  }
+
+  /**
+   * Creates a throw away data directory for Stargate's ephemeral files. It can be overridden using
+   * the system property {@code "stargate.basedir"} which can be useful on systems where the
+   * temporary directory is periodically cleaned.
+   *
+   * @return A file handle to the base directory.
+   * @throws IOException if the base directory is invalid or unable to be created.
+   */
+  protected File getBaseDir() throws IOException {
```

Comment:
nit: maybe this can be part of the `BaseActivator` directly.. No need to extend into a dedicated sub-class.. I know that other modules will not use it now, but in general base directory can be helpful to every module.

Reply:
+1 to moving into `BaseActivator`

Reply:
What advantage does that give us?

Reply:
It may be used later, I guess. It is fairly generic if other modules need it in the future it would be nice to manage the "base" dir in one place.

Reply:
Nm, I just moved it. :)